api/v1alpha4/index/index.go

```diff
@@ -102,7 +90,26 @@ func IndexMachineByProviderID(o client.Object) []string {
 		return nil
 	}
 
-	providerID, err := NewProviderID(*machine.Spec.ProviderID)
+	providerID, err := noderefutil.NewProviderID(*machine.Spec.ProviderID)
+	if err != nil {
+		// Failed to create providerID, skipping.
+		return nil
+	}
+	return []string{providerID.IndexKey()}
+}
+
+// NodeByProviderID contains the logic to index Nodes by ProviderID.
+func NodeByProviderID(o client.Object) []string {
```

Comment:
Should this method be private?

Reply:
Isn't this an index?

Reply:
This method is used in several places when setting up the cluster cache tracker indexes. That is why this method is not private.
May be we can make this private if we setup the `DefaultRemoteIndexes` thing that you are suggesting.